### PR TITLE
Helpers for calibration logic

### PR DESCRIPTION
**Pull request by liviajakob**
Just a bunch of random helper methods for the timeseries calibration logic I'm currently playing around with.

### CODE REVIEW

File: glambie/util/date_helpers.py [hunk 1]
```diff
@@ -84,3 +87,52 @@ def get_year_timedelta(year: int) -> timedelta:
...
+
+    Parameters
+    ----------
+    glaciological_year_start : float
+        a floating point number of when the glaciological year should start (between 0 and 1), e.g. 0.75 for October
+    min_date : float
+        minimum date to be calculated for, in fractional years
+    max_date : float
+        maximum date to be calculated for, in fractional years
+    return_type : str, optional
```
Comment (TristanGoss) [inline glambie/util/date_helpers.py:105]: slight disagreement here, the argument is "returntype" but the docstring says "return_type".

File: glambie/util/date_helpers.py [hunk 2]
```diff
@@ -84,3 +87,52 @@ def get_year_timedelta(year: int) -> timedelta:
...
+        type in which the result is returned. Current options are: 'arrays' and 'dataframe', by default 'arrays'
+
+    Returns
+    -------
+    Union[np.array, np.array] or pd.DataFrame, depending on specified return_type
```
Comment (TristanGoss) [inline glambie/util/date_helpers.py:110]: I think you mean Tuple here rather, than Union

File: glambie/util/timeseries_helpers.py [hunk 3]
```diff
@@ -238,6 +238,29 @@ def timeseries_as_months(fractional_year_array: np.array, downsample_to_month: b
     return monthly_array
 
 
+def timeseries_is_monthly_grid(fractional_year_array: np.array) -> bool:
```
Comment (TristanGoss) [inline glambie/util/timeseries_helpers.py:241]: Ahh, I get it now. I can think of terms other than "monthly grid", but it does succesfully communicate that we're not talking about calendar months.

File: glambie/util/timeseries_helpers.py [hunk 4]
```diff
@@ -447,11 +470,14 @@ def resample_derivative_timeseries_to_monthly_grid(start_dates, end_dates, chang
...
     changes : np.array or list
         changes between start and end date
+    return_type : str, optional
+        type in which the result is returned. Current options are: 'arrays' and 'dataframe', by default 'arrays'
 
     Returns
     -------
-    Union[np.ndarray,np.ndarray,np.ndarray]
-        (start_dates, end_dates, changes) resampled to the monthly grid
+    Union[np.array, np.array, np.array] or pd.DataFrame, depending on specified return_type
```
Comment (TristanGoss) [inline glambie/util/timeseries_helpers.py:478]: Again, I think you mean Tuple here. Union[a, b] means "can be either type a or type b".

File: glambie/util/timeseries_helpers.py [hunk 5]
```diff
@@ -460,4 +486,84 @@ def resample_derivative_timeseries_to_monthly_grid(start_dates, end_dates, chang
...
+    """
+    if return_type == "value":
+        return np.nansum(changes)
+    elif return_type == "dataframe":
+        return pd.DataFrame({"start_dates": [np.nanmin(start_dates)],
+                             "end_dates": [np.nanmax(end_dates)],
+                             "changes": [np.nansum(changes)]})
+
+
+def get_average_trends_over_new_time_periods(start_dates, end_dates, changes, new_start_dates, new_end_dates):
```
Comment (TristanGoss) [inline glambie/util/timeseries_helpers.py:526]: I see - I imagine you've used "trend" since this logic isn't necessarily for any one particular variable.

**Review by TristanGoss — APPROVED** [review]
Some very minor documentation issues here, but I see nothing wrong with the logic, so I'll approve.

File: glambie/util/date_helpers.py (hunk 1 above)
Comment (liviajakob) [reply glambie/util/date_helpers.py:105]: Done

File: glambie/util/date_helpers.py (hunk 2 above)
Comment (liviajakob) [reply glambie/util/date_helpers.py:110]: Done

File: glambie/util/timeseries_helpers.py (hunk 4 above)
Comment (liviajakob) [reply glambie/util/timeseries_helpers.py:478]: Done, also changing everywhere else

File: glambie/util/timeseries_helpers.py (hunk 5 above)
Comment (liviajakob) [reply glambie/util/timeseries_helpers.py:526]: yes, could also use average change I guess. but it's consistent with "get_total_trend"

**Comment by liviajakob** [comment]
Thanks Tristan for the very quick review :)

**Pull request merged**